flink-runtime/src/main/java/org/apache/flink/runtime/execution/Environment.java

```diff
@@ -232,4 +234,20 @@ void acknowledgeCheckpoint(
     IndexedInputGate[] getAllInputGates();
 
     TaskEventDispatcher getTaskEventDispatcher();
+
+    // --------------------------------------------------------------------------------------------
+    //  Fields set in the StreamTask to provide access to mailbox and other runtime resources
+    // --------------------------------------------------------------------------------------------
+
+    default void setMainMailboxExecutor(MailboxExecutor mainMailboxExecutor) {}
+
+    default MailboxExecutor getMainMailboxExecutor() {
```

Comment:
```
Optional<MailboxExecutor> getMainMailboxExecutor() { return Optional.empty(); };
```
? and ditto for `getAsyncOperationsThreadPool()`?
Currently this is unchecked `@Nullable`, and `Optional` would solve compile time checks plus solve the problem with `UnsupportedOperationException()`;

Or maybe keep it `MailboxExecutor getMainMailboxExecutor()` but internally inside the implementation do :
```
public  MailboxExecutor getMainMailboxExecutor() {
  checkState(mailboxExecutor != null, "Trying to access not yet initialized mailbox");
  return mailboxExecutor;
}
```

Reply:
I do not like Optional, I will add a check in `get`.

Reply:
In MockEnvironment, both mailboxExecutor and asycnThreadPool are initialized, and I marked the parameter of `set method` as not nullable. So a `get method` won't return a null;

I will add the check in `RuntimeEnvironment Getter`